### ISSUE TITLE
DockSpaceOverViewport with BeginMainMenuBar?

### ISSUE THREAD

**Issue by philrdubois** [ocornut/imgui#2889]
**Version/Branch of Dear ImGui:**

Version: 1.74 WIP (17301)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl.cpp + imgui_impl_opengl3.cpp 
Compiler: g++
Operating System: linux

**My Issue/Question:**

Hey all,

What a wonderful and useful library! There is however an issue I've been fighting with.

Whenever using the dockspace over viewport feature, use of the main menu bar draws over the tab bar / window decoration. I know there's mention in the code that DockSpaceOverViewport() would prevent the use of BeginMainMenuBar(), however this would seem to be common use so it would be nice if it was implemented. Something such as padding to the top of the viewport for docking would do the trick, however I don't know where to begin to implement such a thing!

Thanks,
Phil.

**Screenshots/Video**
Note that window decorations and tab bars work for other windows:
![Screenshot_2019-11-11_14-49-04](https://user-images.githubusercontent.com/38036563/68620209-a0ef6000-0492-11ea-9edd-9a43c48a77b2.png)

**Standalone, minimal, complete and verifiable example:** 
```
ImGui::DockSpaceOverViewport();
if (ImGui::BeginMainMenuBar()) {
    if (ImGui::BeginMenu("File")) {
        ImGui::EndMenu();
    }
    if (ImGui::BeginMenu("Edit")) {
        ImGui::EndMenu();
    }
    ImGui::EndMainMenuBar();
}
ImGui::ShowAboutWindow();
```

**Comment by ocornut** [comment]
The comment  for `DockspaceOverViewport()` says:
> // Tips: Use with ImGuiDockNodeFlags_PassthruCentralNode!
> // The limitation with this call is that your window won't have a menu bar.
> // Even though we could pass window flags, it would also require the user to be able to call BeginMenuBar() somehow meaning we can't Begin/End in a single function.
> // So if you want a menu bar you need to repeat this code manually ourselves. As with advanced other Docking API, we may change this function signature.

For now you'll probably want to create your own window hosting the dockspace by essentially replicating the code in `DockSpaceOverViewport()`, and:
- either you add a menu to that window (using `BeginMenuBar()` instead of `BeginMainMenuBar()`)
- either you alter the position and size of the window to leave space for the main menu bar.

Ideally we need another solution such as main menu bar communicating their need to take space from the main work area (which would also have the effect that, e.g. you can't lose a thin window under a menu bar).

**Comment by philrdubois** [comment]
Thanks for the rapid response.

I was afraid you'd say that... I had considered the first point there but was worried I'd be going down a rabbit hole. Sobeit, it will give me the opportunity to learn more about OpenGL and this library.

If I find a worthy solution, I'll post it here.

**Comment by ocornut** [comment]
I'm not sure what this has to do with "OpenGL" ? I gave you a solution above, copy `DockSpaceOverViewport()` into your own function. All the calls in there are public ImGui API calls the function does nothing very special.

**Comment by philrdubois** [comment]
Apologies, I must have misunderstood. I have very little graphics experience, and no 3D experience. I was under the impression that it would be more difficult to draw in an ImGui window than in the default viewport, but if that's not the case then disregard my comment.

**Comment by ocornut** [comment]
Closing as duplicate of #3035. Even though #3035 came in later the discussion there is envisioning a possible solution so I'll leave the other open until solved.

**Comment by ocornut** [comment]
Update: this is now fixed by 75de34e281141f851f7e5f9f6a923573167bc0f0 !
